---
title: "dogma20"
output: rmarkdown::pdf_document
vignette: >
  %\VignetteIndexEntry{dogma20}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

```{r setup}
library(dogma20)
```

# Description

## Global description

This package has all the function necessary to take a DNA sequence and analyse the distribution of the output amino acids.

## create_random_seq

This function creates a random sequence of length seq_len

```{r}
DNA_seq <- create_random_seq(12)
DNA_seq
```

## make_RNA_seq

make_RNA_seq is a function that transcribes a DNA sequence into an RNA sequence by replacing thymine (T) with uracil (U).

**Usage:**

```{r}
RNA_seq <- make_RNA_seq(DNA_seq)
RNA_seq
```

-   DNA_seq: A string representing a DNA sequence composed of nucleotide letters A, T, C and G

The function returns a string representing the RNA sequence derived from the input DNA sequence, where thymine (`T`) is replaced by uracil (`U`).

## Split codons

This function takes as input:

-   A string representing a nucleotide sequence - e.g. DNA or RNA.

-   An integer specifying the start position in the sequence - default is 1, meaning the function will start splitting from first position.

The function splits the sequence into strings of three characters (nucleotides) starting from the start position. It returns a vector of the strings representing the codons.

## convert_seq_to_prot

The sequence takes a list of codons as an input and converts it into a protein sequence.

```{r}
prot_seq = convert_seq_to_prot(codons_sequence)
prot_seq
```

## plotting_amino_acid_distribution

This function counts how much we have of each amino acids and plots the result.

```{r}
p = plotting_amino_acid_distribution(prot_seq)
p
```

# Use cases

# Discussion
